### PR TITLE
feat: adjusted interfaces and api methods for hub api changes

### PR DESCRIPTION
**Pull request by tada5hi**

### CODE REVIEW

File: tests/test_hub.py [hunk 1]
```diff
@@ -37,15 +36,6 @@ def _check_buckets_exist():
 
     assert eventually(_check_buckets_exist)
 
-    # check that buckets are listed correctly
```
Comment (mjugl) [inline tests/test_hub.py:40]: Please add this back. Removing this just breaks the test.

File: project/hub.py [hunk 2]
```diff
@@ -241,9 +248,9 @@ def get_bucket_list(self) -> ResourceList[Bucket]:
         r.raise_for_status()
         return ResourceList[Bucket](**r.json())
 
-    def get_bucket_by_id_or_name(self, bucket_id_or_name: str | UUID) -> Bucket | None:
```
Comment (mjugl) [inline project/hub.py:244]: Any reason why this was renamed? From my end I can still get buckets by either their ID or name just fine.

File: tests/test_hub.py [hunk 3]
```diff
@@ -73,16 +63,18 @@ def uploaded_bucket_file(result_bucket_name, api_client, rng):
 def test_link_bucket_file_to_analysis(uploaded_bucket_file, analysis_id, api_client):
     _, bucket_file = uploaded_bucket_file
 
+    analysis_bucket = api_client.get_analysis_bucket(analysis_id, "RESULT")
+
     # check that the analysis file was created
     analysis_file = api_client.link_bucket_file_to_analysis(
-        analysis_id, bucket_file.id, bucket_file.name, bucket_type="RESULT"
+        analysis_bucket.external_id, bucket_file.id, bucket_file.name
```
Comment (mjugl) [inline tests/test_hub.py:70]: Might need some clarification on what the new `/analysis-bucket-files` endpoint *really* does. It seems as if the parameter order is off but flipping them doesn't fix it. And with my current understanding, `external_id` and `bucket_id` refer to the same thing. Where is the link to the analysis?

**Review by mjugl — CHANGES_REQUESTED** [review]
At least the test setup needs to be fixed. That gives 3/4 passing tests. The failing `test_link_bucket_file_to_analysis` test will require some extra explanation because it's not clear to me how it actually links back to the analysis.

Other parts will be reviewed when the Hub client tests pass.

File: tests/test_hub.py (hunk 1 above)
Comment (tada5hi) [reply tests/test_hub.py:40]: the name property of the bucket in the storage service will no longer be structured according to a fixed schema. The comparison should therefore no longer take place.

File: tests/test_hub.py (hunk 3 above)
Comment (tada5hi) [reply tests/test_hub.py:70]: The analysis-bucket-files endpoint is not a new endpoint. The endpoint /analysis-files has been renamed to the endpoint /analysis-bucket-files. The external_id refers to the identifier in the storage service.
The endpoint /analysis-buckets has been added to store a specific analysis and a type (temp, result, code), as well as the external_id (bucket identifier in the storage service)

File: project/hub.py (hunk 2 above)
Comment (tada5hi) [reply project/hub.py:244]: because a bucket should only be requested by its unique id and no longer by its name

**Review by mjugl — APPROVED** [review]

**Pull request merged**